crates/chain/src/tx_graph.rs

```diff
@@ -319,14 +319,19 @@ impl<A> TxGraph<A> {
             .map(|(outpoint, spends)| (outpoint.vout, spends))
     }
 
-    /// Creates an iterator of ancestor transactions from the starting `Transaction` up to a
-    /// specified depth limit, not including the original `Transaction`.
-    pub fn walk_ancestors<'g>(
+    /// Creates an iterator of unconfirmed ancestor transactions from the starting `Transaction`,
```

Comment:
`walk_ancestors` should traverse all ancestors. It sounds like a general method traversing ancestors in a graph. You could have another method for `walk_unconfirmed_ancestors`.

Reply:
The next step, after discussion with @evanlinjin, was to refactor `TxAncestors` again to take in a closure for versatility (similar to `TxDescendants`), which will allow `walk_ancestors` to traverse all ancestors. The work for this is nearly complete.